packages/config/eslint-preset.js

```diff
@@ -23,6 +23,7 @@ module.exports = {
     "@next/next/no-html-link-for-pages": "off",
     "jsx-a11y/role-supports-aria-props": "off", // @see https://github.com/vercel/next.js/issues/27989#issuecomment-897638654
     "playwright/no-page-pause": "error",
+    "@typescript-eslint/no-floating-promises": "error",
```

Comment:
Enabling `no-floating-promises` without making the above change throws the following error.

```Error while loading rule '@typescript-eslint/no-floating-promises': You have used a rule which requires parserServices to be generated. You must therefore provide a value for the "parserOptions.project" property for @typescript-eslint/parser.```

I found the problem after some debugging to be that the `tsconfigRootDir` being set to `__dirname` is causing the absolute path for `./apps/*/tsconfig.json` to become `packages/config/apps/*/tsconfig.json` which is wrong. So, I fixed the path accordingly.

But after that I am seeing that there is Heap out of Memory error now. Tried with even `--max-old-space-size=8096`
<img width="951" alt="Screenshot 2023-08-24 at 3 32 15 PM" src="https://github.com/calcom/cal.com/assets/1780212/a9374a5a-1a8f-4d7f-b840-97e1c00598c2">

Reply:
I think you're only just now running into this issue as this is the first typed rule. [Many such cases](https://github.com/typescript-eslint/typescript-eslint/issues/1192) for large monorepos.

Might make sense to move `parserOptions` and typed rules to the TS override. We can set `tsconfigRootDir` to the repo root and `project: true`. ESLint will then automatically pick the tsconfig nearest each file being processed.

```diff
module.exports = {
  ...
- parserOptions: {
-   tsconfigRootDir: __dirname,
-   project: ["../../apps/*/tsconfig.json", "../../packages/*/tsconfig.json"],
- },
  ...
  rules: {
    "@next/next/no-img-element": "off",
    "@next/next/no-html-link-for-pages": "off",
    "jsx-a11y/role-supports-aria-props": "off", // @see https://github.com/vercel/next.js/issues/27989#issuecomment-897638654
    "playwright/no-page-pause": "error",
-   "@typescript-eslint/no-floating-promises": "error",
    ...
  },
  overrides: [
    {
      files: ["*.ts", "*.tsx"],
      extends: ["plugin:@typescript-eslint/recommended", "plugin:@calcom/eslint/recommended"],
      plugins: ["@typescript-eslint", "@calcom/eslint"],
      parser: "@typescript-eslint/parser",
+     parserOptions: {
+       tsconfigRootDir: require("path").resolve(__dirname, "..", ".."),
+       project: true,
+     },
      rules: {
+        "@typescript-eslint/no-floating-promises": "error",
        ...
```

This worked for me, but there is also an alternative if `@typescript-eslint/parser` is upgraded. [This new experimental flag](https://github.com/typescript-eslint/typescript-eslint/pull/6754) works quite well.

Reply:
Oh thanks so much @nicktrn. I will look into it.